### ISSUE TITLE
example of getLastTransactionID()

### ISSUE THREAD

**Issue by kiros32** [PHPMailer/PHPMailer#1123]
Guys , can anyone provide real example - how to use getLastTransactionID() method after sending message to postfix?

```
$mail->send();

$tid = $mail->getSMTPInstance()->getLastTransactionID();
var_dump($tid); // false

$reply = $mail->getSMTPInstance()->getLastReply();
var_dump($reply); // "250 2.0.0 Ok "
```

debug info is
```
Fri, 11 Aug 2017 07:30:43 +0000 | SMTP -> get_lines(): $data is ""
Fri, 11 Aug 2017 07:30:43 +0000 | SMTP -> get_lines(): $str is  "250 2.0.0 Ok: queued as 38C6F2208F
"
Fri, 11 Aug 2017 07:30:43 +0000 | SERVER -> CLIENT: 250 2.0.0 Ok: queued as 38C6F2208F

Fri, 11 Aug 2017 07:30:43 +0000 | CLIENT -> SERVER: RSET

Fri, 11 Aug 2017 07:30:43 +0000 | SMTP -> get_lines(): $data is ""
Fri, 11 Aug 2017 07:30:43 +0000 | SMTP -> get_lines(): $str is  "250 2.0.0 Ok
"
Fri, 11 Aug 2017 07:30:43 +0000 | SERVER -> CLIENT: 250 2.0.0 Ok

Fri, 11 Aug 2017 07:30:43 +0000 | CLIENT -> SERVER: QUIT

Fri, 11 Aug 2017 07:30:43 +0000 | SMTP -> get_lines(): $data is ""
Fri, 11 Aug 2017 07:30:43 +0000 | SMTP -> get_lines(): $str is  "221 2.0.0 Bye
"
Fri, 11 Aug 2017 07:30:43 +0000 | SERVER -> CLIENT: 221 2.0.0 Bye

Fri, 11 Aug 2017 07:30:43 +0000 | Connection: closed
```

**Comment by Synchro** [comment]
In this case your postfix server is not providing a message ID, so PHPMailer can't extract one. As the docs say:

> If no reply has been received yet, it will return null.
> If no pattern has been matched, it will return false.

You'll need to read postfix documentation to figure out why your server isn't emitting queue IDs here.

**Comment by kiros32** [comment]
NO

I am not talking about Message ID , I am talking about transactionid

```SERVER -> CLIENT: 250 2.0.0 Ok: queued as 38C6F2208F```

38C6F2208F is transactionid

The method getLastTransactionID , I think , created for getting this transactionid

BUT
it analyses last reply
and `SERVER -> CLIENT: 250 2.0.0 Ok: queued as 38C6F2208F` is not a last reply

**Comment by kiros32** [comment]
@Synchro any ideas?

**Comment by Synchro** [comment]
I wrote a reply earlier but a power failure lost it...

I know it's not a message ID - I just used the wrong term in that last comment, however, `getLastTransactionID` *does* retrieve the last queue/transaction ID and not the message ID. The problem is that when you have `SMTPKeepAlive` set, PHPMailer sends a `RSET` command after the `DATA` command, and this means that the reply to the `DATA` command is lost in favour of the reply to `RSET`. I've made a fix in the 6.0 branch - can you give it a try?

**Comment by kiros32** [comment]
@Synchro yes , it works now! Thanks! Merge it to stable branch pls
